subsys/net/lib/wifi_ready/wifi_ready.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/net/net_event.h>
+#include <supp_events.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <net/wifi_ready.h>
+
+LOG_MODULE_REGISTER(wifi_ready, CONFIG_WIFI_READY_EVENT_HANDLING_LOG_LEVEL);
+
+static app_callback_t callback;
```

Comment:
What about multiple applications registering the callback?

Reply:
Added a check for the time being. Will handle fully at a later point in time.